services/src/main/java/com/linkedin/kmf/common/Utils.java

```diff
@@ -35,13 +54,178 @@
+      int partitionFactor) {
+    ZkUtils zkUtils = ZkUtils.apply(zkUrl, ZK_SESSION_TIMEOUT_MS, ZK_CONNECTION_TIMEOUT_MS, JaasUtils.isZkSecurityEnabled());
+    try {
+      if (AdminUtils.topicExists(zkUtils, topic)) {
+        LOG.info("Monitoring topic \"" + topic + "\" already exists.");
+        return getPartitionNumForTopic(zkUrl, topic);
+      }
+
+      int brokerCount = zkUtils.getAllBrokersInCluster().size();
+
+      if (partitionFactor <= 0) {
+        throw new IllegalArgumentException("Partition factor must be greater than zero, but was configured for " +
+            partitionFactor + ".");
+      }
+      int partitionCount = brokerCount * partitionFactor;
+
+      int minIsr = Math.max(replicationFactor - 1, 1);
+      Properties topicConfig = new Properties();
+      topicConfig.setProperty(KafkaConfig.MinInSyncReplicasProp(), Integer.toString(minIsr));
+      AdminUtils.createTopic(zkUtils, topic, partitionCount, replicationFactor, new Properties());
+
+      LOG.info("Created monitoring topic \"" + topic + "\" with " + partitionCount + " partitions, min ISR of " + minIsr
+          + " and replication factor of " + replicationFactor + ".");
+
+      return partitionCount;
+    } finally {
+      zkUtils.close();
+    }
+  }
+
+  /**
+   * This is so you can call KafkaConsumer.partitionsFor don't consume messages from this that is the job of the
+   * ConsumeService.
+   *
+   * @param brokerList probably get this from the configuration file.
+   * @param zkUrl zoo keeper url
+   * @param topic the monitoring topic
+   * @return an initialized KafkaConsumer
+   */
+
+  private static KafkaConsumer<?,?> constructClientForMetadata(String brokerList, String zkUrl, String topic) {
+
+    Properties consumerProps = new Properties();
+
+    // Assign default config. This has the lowest priority.
+    consumerProps.put(ConsumerConfig.ENABLE_AUTO_COMMIT_CONFIG, "false");
+    consumerProps.put(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "latest");
+    consumerProps.put(ConsumerConfig.CLIENT_ID_CONFIG, "kmf-metadata-consumer");
+    consumerProps.put(ConsumerConfig.GROUP_ID_CONFIG, "kmf-metadata-consumer-group");
+    consumerProps.put(ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+    consumerProps.put(ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class.getName());
+
+    // Assign config specified for ConsumeService.
+    consumerProps.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG, brokerList);
+    consumerProps.put("zookeeper.connect", zkUrl);
+
+    KafkaConsumer<String, String> metadataConsumer = new KafkaConsumer<>(consumerProps);
+    metadataConsumer.subscribe(Collections.singletonList(topic));
+    return metadataConsumer;
+  }
+
+  /**
+   *
+   * @param zkUrl the zookeeper url
+   * @param topic the monitored topic
+   * @param partitionThreshold  This assumes that you want to have at least one per broker.  When the number of partitions
```

Comment:
I'm going to remove this code from this pull request and we can discuss it in a different pull request.